Sources/TuistKit/Services/Project/ProjectTokensListService.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import Path
+import TuistLoader
+import TuistServer
+import TuistSupport
+
+protocol ProjectTokensListServicing {
+    func run(
+        fullHandle: String,
+        directory: String?
+    ) async throws
+}
+
+final class ProjectTokensListService: ProjectTokensListServicing {
+    private let listProjectTokensService: ListProjectTokensServicing
+    private let serverURLService: ServerURLServicing
+    private let configLoader: ConfigLoading
+
+    init(
+        listProjectTokensService: ListProjectTokensServicing = ListProjectTokensService(),
+        serverURLService: ServerURLServicing = ServerURLService(),
+        configLoader: ConfigLoading = ConfigLoader()
+    ) {
+        self.listProjectTokensService = listProjectTokensService
+        self.serverURLService = serverURLService
+        self.configLoader = configLoader
+    }
+
+    func run(
+        fullHandle: String,
+        directory: String?
+    ) async throws {
+        let directoryPath: AbsolutePath
+        if let directory {
+            directoryPath = try AbsolutePath(validating: directory, relativeTo: FileHandler.shared.currentPath)
+        } else {
+            directoryPath = FileHandler.shared.currentPath
+        }
+        let config = try configLoader.loadConfig(path: directoryPath)
+        let serverURL = try serverURLService.url(configServerURL: config.url)
+
+        let tokens = try await listProjectTokensService.listProjectTokens(
+            fullHandle: fullHandle,
+            serverURL: serverURL
+        )
+
+        if tokens.isEmpty {
+            logger.notice("No project tokens found. Create one by running `tuist project tokens create \(fullHandle).")
```

Comment:
I could derive `tokens create` relatively easily by:
```swift
"\(ProjectTokensListCommand.configuration.commandName) \(ProjectTokensListCommand.configuration._superCommandName)"
```

Getting the rest is a bit more tricky as from the `_superCommandName`, there's no easy way to get the type, so we could traverse the super commands up. I think we'll need to skip this, although I would agree it would be amazing if there was a more-or-less straightforward way to achieve that.